spec/models/user_garden_spec.rb

```diff
@@ -0,0 +1,16 @@
+require 'rails_helper'
+
+RSpec.describe UserGarden, type: :model do
+  describe 'relationships' do
+    it { should belong_to :user}
+    it { should belong_to :garden }
+  end
+
+  describe 'creation' do
+    it 'can create user garden' do
+      user = User.create!(email: 'test@gmail.com', provider: 'google', token: '1234')
```

Comment:
maybe we could make the provider an enum as well? I think storing as an integer is more efficient than as a string

Reply:
Agreed. However, maybe for now we focus on one oAuth and later add more we can change to enum?